File: src/main/java/org/example/scalingstream/partitioner/RoundRobinPartitioner.java
package org.example.scalingstream.partitioner;

import org.example.scalingstream.Record;
import org.example.scalingstream.operator.OutputBuffers;

public class RoundRobinPartitioner extends Partitioner {
    private int i = 0;

    RoundRobinPartitioner(int numOut) {
        super(numOut);
        this.i = 0;
    }

    @Override
    public int assignPartition(Record record) {
        return (i++) % numOut;
    }

    @Override
    public void assignPartition(OutputBuffers outputBuffers, Record[] records) {
        for (Record record: records) {
            outputBuffers.append(assignPartition(record), record);
        }
    }
}

File: src/main/java/org/example/scalingstream/partitioner/RoundRobinPartitioner.kt
package org.example.scalingstream.partitioner

class RoundRobinPartitioner : Partitioner {
    override val name: String = "ROUND_ROBIN"

    private var i = 0
    override fun <Type> assignPartition(record: Type, numOut: Int): Int {
        i = ++i % numOut
        return i
    }
}
